lib/auth/auth.go

```diff
@@ -3569,6 +3569,16 @@ func (a *Server) PingInventory(ctx context.Context, req proto.InventoryPingReque
 	}, nil
 }
 
+// UpdateLabels updates the labels on an instance over the inventory control
+// stream.
+func (a *Server) UpdateLabels(ctx context.Context, req proto.InventoryUpdateLabelsRequest) error {
+	stream, ok := a.inventory.GetControlStream(req.ServerID)
+	if !ok {
+		return trace.NotFound("no control stream found for server %q", req.ServerID)
+	}
+	return trace.Wrap(stream.UpdateLabels(ctx, req.Labels))
+}
+
```

Comment:
note: you may already be aware of this, but its the kind of gotcha that is easy to accidentally accidentally overlook... agents create exactly one inventory control stream each, meaning that they are divided among however many auths are in the cluster.  E.g. if there are 2 auths, each auth has a control stream to half the agents.

Having methods that act directly on a live control stream can be useful for testing/debugging, but are finicky to use in practice since you need to be running `tctl` locally on whichever auth happens to be connected to the agent you care about.

Reply:
Eventually there will be a reconciler service calling this on each auth server, so as long as the right stream is present on at least one of them, I think this should be fine.

Also see [this comment](https://github.com/gravitational/teleport/pull/26112#discussion_r1202936000) with respect to `tctl`.